tfhe/src/shortint/server_key/tests/noise_level.rs

```diff
@@ -0,0 +1,263 @@
+use crate::shortint::ciphertext::NoiseLevel;
+use crate::shortint::keycache::KEY_CACHE;
+use crate::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+use crate::shortint::{Ciphertext, ServerKey};
+
+fn test_1_ct_noise_level_propagation(sk: &ServerKey, ct: &Ciphertext) {
+    let test_fn = |f: &dyn Fn(&ServerKey, &Ciphertext) -> Ciphertext,
+                   g: &dyn Fn(NoiseLevel) -> NoiseLevel| {
+        assert!(f(sk, ct).noise_level == g(ct.noise_level));
+    };
+
+    test_fn(&ServerKey::unchecked_neg, &|a| a);
+    test_fn(
+        &|sk, ct| ServerKey::unchecked_neg_with_correcting_term(sk, ct).0,
+        &|a| a,
```

Comment:
maybe rename a to "noise_is_unchanged" or "input_ct_noise" so that reading the test is made easier, as here we don't have a Noise expression (same thing elsewhere)

It's a detail but should help readability if ever we are down there debugging noise stuff

Reply:
Renamed `ct_noise`